src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,176 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exeptions.UserRegistrationException;
+import core.basesyntax.model.User;
+import java.time.LocalDateTime;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "test_login_created_at " + LocalDateTime.now();
+    private static final String VALID_PASSWORD = "Mine!$VAl1DP@ZSW0Rd!)(*&^%$#@";
+    private static final int VALID_AGE = 20;
+    private static final String VALID_EDGE_LOGIN = "testLo";
+    private static final String VALID_EDGE_PASSWORD = "MiP@33";
+    private static final int VALID_EDGE_AGE = 18;
+    private static final String INVALID_SHORT_LOGIN = "testL";
+    private static final String INVALID_SHORT_PASSWORD = "MiP@3";
+    private static final int INVALID_AGE = 17;
+    private static final String INVALID_EMPTY_LOGIN = "";
+    private static final String INVALID_EMPTY_PASSWORD = "";
+    private static final int INVALID_ZERO_AGE = 0;
+
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+    private User user;
+
+    @BeforeEach
+    public void setUp() {
+        user = new User(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+    }
+
+    @AfterEach
+    void noWitnesses() {
+        Storage.people.clear();
+    }
+
+    @Test
+    public void register_nullLogin_notOK() {
+        user.setLogin(null);
+        assertThrows(UserRegistrationException.class, () -> {
+            registrationService.register(user);
+        },"Login field cannot be null");
+    }
+
+    @Test
+    public void register_nullPass_notOK() {
+        user.setPassword(null);
+        assertThrows(UserRegistrationException.class, () -> {
+            registrationService.register(user);
+        },"Password field cannot be null");
+    }
+
+    @Test
+    public void register_nullAge_notOK() {
+        user.setAge(null);
+        assertThrows(UserRegistrationException.class, () -> {
+            registrationService.register(user);
+        },"Age field cannot be null");
+    }
+
+    @Test
+    public void register_NullLoginExeption_notOk() {
+        user.setLogin(null);
+        try {
+            registrationService.register(user);
+        } catch (UserRegistrationException e) {
+            return;
+        }
+        fail("Password cannot be null, exception about login check failed should see you");
+    }
+
+    @Test
+    public void register_NullPassExeption_notOk() {

```

Comment:
we should avoid shortened names and abbreviations
```suggestion
    public void register_NullPasswordException_notOk() {

```